File: module-app/src/main/java/maple/expectation/controller/dto/auth/TokenResponse.java
package maple.expectation.controller.dto.auth;

/**
 * Token Refresh 응답 DTO (Issue #279)
 *
 * @param accessToken 새 JWT Access Token
 * @param accessExpiresIn Access Token 만료 시간 (초)
 * @param refreshToken 새 Refresh Token ID
 * @param refreshExpiresIn Refresh Token 만료 시간 (초)
 */
public record TokenResponse(
    String accessToken, Long accessExpiresIn, String refreshToken, Long refreshExpiresIn) {

  /**
   * 정적 팩토리 메서드 (TokenResponse 생성)
   *
   * @param accessToken 새 JWT Access Token
   * @param accessExpiresIn Access Token 만료 시간 (초)
   * @param refreshToken 새 Refresh Token ID
   * @param refreshExpiresIn Refresh Token 만료 시간 (초)
   * @return TokenResponse 인스턴스
   */
  public static TokenResponse of(
      String accessToken, Long accessExpiresIn, String refreshToken, Long refreshExpiresIn) {
    return new TokenResponse(accessToken, accessExpiresIn, refreshToken, refreshExpiresIn);
  }
}

File: module-web/src/main/kotlin/maple/expectation/web/dto/TokenResponse.kt
package maple.expectation.web.dto

/**
 * Token Refresh 응답 DTO (Issue #279)
 *
 * @property accessToken 새 Access Token (JWT)
 * @property accessExpiresIn Access Token 만료 시간 (초)
 * @property refreshToken 새 Refresh Token ID
 * @property refreshExpiresIn Refresh Token 만료 시간 (초)
 */
data class TokenResponse(
    val accessToken: String?,
    val accessExpiresIn: Long?,
    val refreshToken: String?,
    val refreshExpiresIn: Long?,
) {
    // Record-style accessors for production code compatibility
    fun accessToken(): String = accessToken ?: ""
    fun accessExpiresIn(): Long = accessExpiresIn ?: 0
    fun refreshToken(): String = refreshToken ?: ""
    fun refreshExpiresIn(): Long = refreshExpiresIn ?: 0

    // JavaBean-style getters for Java tests (handle null from mocks)
    @JvmName("getAccessTokenSafe")
    fun getAccessToken(): String = accessToken ?: ""

    @JvmName("getAccessExpiresInSafe")
    fun getAccessExpiresIn(): Long = accessExpiresIn ?: 0

    @JvmName("getRefreshTokenSafe")
    fun getRefreshToken(): String = refreshToken ?: ""

    @JvmName("getRefreshExpiresInSafe")
    fun getRefreshExpiresIn(): Long = refreshExpiresIn ?: 0

    companion object {
        @JvmStatic
        fun of(
            accessToken: String?,
            accessExpiresIn: Long?,
            refreshToken: String?,
            refreshExpiresIn: Long?,
        ): TokenResponse = TokenResponse(accessToken, accessExpiresIn, refreshToken, refreshExpiresIn)
    }
}
